test/controllers/users_controller_test.rb

```diff
@@ -1,5 +1,39 @@
 require 'test_helper'
 
 describe UsersController do
+  describe "index" do
+    it "goes to index view" do
 
+      get users_path
+
+      assert_response :success
+    end
+
+    it "renders users" do
+      users = User.all
+
+      get users_path
+
+      expect( response.body ).must_include("dan")
+      expect( response.body ).must_include("kari")
```

Comment:
This is more work than we typically need to do for controller tests - you're looking at the rendered HTML, and making sure it includes these two values. If your fixture data or view code changes, this test will also need to change.